gadgets/src/less_than.rs

```diff
@@ -0,0 +1,138 @@
+//! Lt chip can be used to compare LT for two expressions LHS and RHS.
+
+use std::array;
+
+use array_init::array_init;
+use eth_types::Field;
+use halo2_proofs::{
+    arithmetic::FieldExt,
+    circuit::{Chip, Region},
+    plonk::{Advice, Column, ConstraintSystem, Error, Expression, VirtualCells},
+    poly::Rotation,
+};
+
+use super::{
+    bool_check,
+    util::{expr_from_bytes, pow_of_two},
+};
+
+/// Instruction that the Lt chip needs to implement.
+pub trait LtInstruction<F: FieldExt> {
+    /// Assign the lhs and rhs witnesses to the Lt chip's region.
+    fn assign(
+        &self,
+        region: &mut Region<'_, F>,
+        offset: usize,
+        lhs: F,
+        rhs: F,
+    ) -> Result<(), Error>;
+}
+
+/// Config for the Lt chip.
+#[derive(Clone, Debug)]
+pub struct LtConfig<F, const N_BYTES: usize> {
+    /// Denotes the lt outcome. If lhs < rhs then lt == 1, otherwise lt == 0.
+    pub lt: Column<Advice>,
+    /// Denotes the bytes representation of the difference between lhs and rhs.
+    pub diff: [Column<Advice>; N_BYTES],
```

Comment:
Is that intended to not check the range of each byte in this config (user should do that somewhere else)? If so, it'd be nice to add some note here.

```suggestion
    /// Denotes the bytes representation in little-endian of the difference between lhs and rhs.
    /// Note that the range of each byte is not checked by this config.
    pub diff: [Column<Advice>; N_BYTES],
```